### ISSUE TITLE
zookeeper in RC not wokring

### ISSUE THREAD

**Issue by schadr** [vmware/photon#452]
when I try starting is through systemctl I see the following lines repeated

```
Apr 28 16:33:03 esxcloud-management-vm systemd[1]: zookeeper.service: Service hold-off time over, scheduling restart.
Apr 28 16:33:03 esxcloud-management-vm systemd[1]: Stopped Apache ZooKeeper.
-- Subject: Unit zookeeper.service has finished shutting down
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit zookeeper.service has finished shutting down.
Apr 28 16:33:03 esxcloud-management-vm systemd[1]: Starting Apache ZooKeeper...
-- Subject: Unit zookeeper.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit zookeeper.service has begun starting up.
Apr 28 16:33:03 esxcloud-management-vm zkServer.sh[19729]: ZooKeeper JMX enabled by default
Apr 28 16:33:03 esxcloud-management-vm zkServer.sh[19729]: Using config: /etc/zookeeper/zoo.cfg
Apr 28 16:33:03 esxcloud-management-vm zkServer.sh[19729]: Starting zookeeper ... /usr/bin/zkServer.sh: line 140: ./zookeeper.out: Permission denied
Apr 28 16:33:04 esxcloud-management-vm zkServer.sh[19729]: STARTED
Apr 28 16:33:04 esxcloud-management-vm zkServer.sh[19743]: ZooKeeper JMX enabled by default
Apr 28 16:33:04 esxcloud-management-vm zkServer.sh[19743]: Using config:
Apr 28 16:33:04 esxcloud-management-vm zkServer.sh[19743]: grep: : No such file or directory
Apr 28 16:33:04 esxcloud-management-vm zkServer.sh[19743]: mkdir: cannot create directory '': No such file or directory
Apr 28 16:33:04 esxcloud-management-vm zkServer.sh[19743]: Stopping zookeeper ... no zookeeper to stop (could not find file /zookeeper_server.pid)
Apr 28 16:33:04 esxcloud-management-vm systemd[1]: Started Apache ZooKeeper.
-- Subject: Unit zookeeper.service has finished start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit zookeeper.service has finished starting up.
-- 
-- The start-up result is done.
```

when running zookeeper from the command line using `zkServer.sh start-foreground /path/to/zoo.cfg` I get the following output:

```
ZooKeeper JMX enabled by default
Using config: zoo.cfg
Error: Could not find or load main class org.apache.zookeeper.server.quorum.QuorumPeerMain
``` 

running `zkServer.sh print-cmd` shows the following:

```
ZooKeeper JMX enabled by default
Using config: 
grep: : No such file or directory
mkdir: cannot create directory '': No such file or directory
"/usr/java/default/bin/java" -Dzookeeper.log.dir="." -Dzookeeper.root.logger="INFO,CONSOLE" -cp "/etc/zookeeper/../build/classes:/etc/zookeeper/../build/lib/*.jar:/etc/zookeeper/../lib/*.jar:/etc/zookeeper/../zookeeper-*.jar:/etc/zookeeper/../src/java/lib/*.jar::"  -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.local.only=false org.apache.zookeeper.server.quorum.QuorumPeerMain "" > "./zookeeper.out" 2>&1 < /dev/null
```

it looks like the classpath is a bit mangled and does not point to the actual zookeeper jar it tries to run.

**Comment by schadr** [comment]
The problem seems to be that zkEnv.sh is completely broken and assume as very specific folder structure that PhotonOS is simply not providing.

```
CLASSPATH="$ZOOCFGDIR:$CLASSPATH"

for i in "$ZOOBINDIR"/../src/java/lib/*.jar
do
    CLASSPATH="$i:$CLASSPATH"
done
```
and
```
if [ -e "${ZOOKEEPER_PREFIX}"/share/zookeeper/zookeeper-*.jar ]; then
  LIBPATH=("${ZOOKEEPER_PREFIX}"/share/zookeeper/*.jar)
else
  #release tarball format
  for i in "$ZOOBINDIR"/../zookeeper-*.jar
  do
    CLASSPATH="$i:$CLASSPATH"
  done
  LIBPATH=("${ZOOBINDIR}"/../lib/*.jar)
fi

for i in "${LIBPATH[@]}"
do
    CLASSPATH="$i:$CLASSPATH"
done
```
are building the classpath, but none of the found folders contain any of the relevant jars. The `zookeeper*.jar` is located in `/usr/lib/zookeeper-3.4.8.jar` and all other dependencies are places in `/usr/lib/`, which is a potential problem since this can easily lead to including multiple library version on a class path.

**Comment by dthaluru** [comment]
I am looking in to this issue.

**Comment by dthaluru** [comment]
Fix is available from commit@bf8c5d2
